src/app/core/state/state.service.ts

```diff
@@ -52,10 +52,16 @@ export class StateService {
     return state;
   }
 
-  observe(prop: string) {
+  observe(prop: string, subkey?: string) {
+    if (subkey) {
+      // TODO: maybe check if subkey exists?
+      // e.g observe('getblockchaininfo', 'blocks') will return only the 'blocks' key from the output.
+      return this._getObservablePair(prop).observable.map(key => key[subkey]);
```

Comment:
If a subkey is provided in the observe(), then the observable.map(..) will only emit a value if the subkey changes.

```
observe('getblockchaininfo', 'blocks')
```
should only cause updates if blocks is updated.

Currently _every_ getwalletinfo/... will cause an update on 'blocks' whether it has actually changed or not..
`.distinctUntilChanged()` would fix this.

Reply:
It only emits if the value of blocks actually changes... Now it will always emit, on every single call

Reply:
Now you have to add a distinct on every observe, seems overkill.. It's still creating an observable for every key, so all this has accomplished is extra code and extra checks, using more cpu than before.

Reply:
You could return the observable with distinctUntilChanged(), as well as to possibly pass in the take until variable, which would reduce the code duplication everytime you use it.

Reply:
@rynomster 
It was already emitting on every update, distinct or not. Take a look at the debug console, in the old code the 'blocks' were updating whether there was an actual change of value or not.. On top of it, it was replaying the whole stored set.

It uses the existing observables and returns a new one mapped to the existing one, I don't think cpu usage will change significantly.

adding distinctUntilChanged() :+1: 

We can't add `takeWhile(() => !this.destroyed)` here because that would only destroy the subscriptions if the _rpc state service is destroyed_, and not when the component is destroyed. The result would be the same as takeWhile(...) nowhere.

Reply:
You can pass in this.destroyed, that way it has context, and actually it was only replaying up until the last time it was used.. each time something is no longer used, it drops the history.
It was only emitting when the value changed. you can go put a debug message on master and you will see, only when a block changes, would it emit that the block has changed